### PR TITLE
make parsing safer

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt [hunk 1]
```diff
@@ -37,12 +35,8 @@ suspend fun parseDirectionsResponse(
             } else {
                 ExpectedFactory.createValue(routes)
             }
-        } catch (ex: Exception) {
-            when (ex) {
-                is JSONException,
-                is JsonSyntaxException -> ExpectedFactory.createError(ex)
-                else -> throw ex
-            }
+        } catch (ex: Throwable) {
```
Comment (DzmitryFomchyn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:38]: Let's add logs here

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt [hunk 2]
```diff
@@ -66,12 +60,7 @@ fun parseRouteInterfaces(
...
             .let { ExpectedFactory.createValue(it) }
-    } catch (ex: Exception) {
-        when (ex) {
-            is JSONException,
-            is IllegalStateException,
-            is IllegalArgumentException -> ExpectedFactory.createError(ex)
-            else -> throw ex
-        }
+    } catch (ex: Throwable) {
+        ExpectedFactory.createError(ex)
```
Comment (DzmitryFomchyn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:64]: Let's add logs here

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt [hunk 3]
```diff
@@ -135,6 +135,8 @@ class NavigationRoute internal constructor(
          *
          * @param directionsResponseJson response to be parsed into [NavigationRoute]s
          * @param routeRequestUrl URL used to generate the [directionsResponse]
+         *
+         * @throws Throwable if `directionsResponseJson` is invalid
```
Comment (DzmitryFomchyn) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:139]: What type of Throwable it throws? Can it ever be `Error`, not `Exception`?

**Review by DzmitryFomchyn — APPROVED** [review]

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt (hunk 3 above)
Comment (dzinad) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:139]: It's supposed to be exception I guess, but it all comes down to gson library, so I don't want to promise anything in their name.

**Review by VysotskiVadim — APPROVED** [review]
Maybe it's worth testing invalid response in the continuous alternatives instrumentation tests?

**Comment by dzinad** [comment]
> Maybe it's worth testing invalid response in the continuous alternatives instrumentation tests?

Done.

**Pull request merged**